src/plugin/ipc/file/filewrappers.cpp

```diff
@@ -371,7 +371,15 @@ static int _open_open64_work(int(*fn) (const char *path, int flags, ...),
     newpath = currPtsDevName;
   }
 
-  int fd = (*fn) (newpath, flags, mode);
+  dmtcp::string hook_path = "";
+  bool doSwap = pathvirt_get_physical_path ? pathvirt_get_physical_path(newpath, hook_path) : false;
+  int fd = -1;
+
+  /* hook_path was NULL, not swapping */
+  if (!doSwap)
+    fd = (*fn)(newpath, flags, mode);
+  else
+    fd = (*fn)(hook_path.c_str(), flags, mode);
 
   if (fd >= 0 && dmtcp_is_running_state()) {
     FileConnList::instance().processFileConnection(fd, newpath, flags, mode);
```

Comment:
@mossberg,
    Wouldn't this code be conceptually simpler if we re-named `pathvirt_get_physical_path(newpath, dmtcp::string hook_path)` to be:
```

char * virtual_to_physical_path(char *newpath, dmtcp::string phys_path);
```

where the usage would be simplified to:
```
dmtcp::string phys_path_string = "");
char *phys_path = virtual_to_physical_path(newpath, phys_path_string);
...
fd = (*fn)(phys_path, flags, mode);
```
In this case, we would define `virtual_to_physical_path(...)` to return either `newpath` (if no swap) or `phys_path_string.c_string()` (if there is a swap).

-----------------------------

If you agree with the above comment, could you re-define/re-name the functions and variables as indicated?  (Sorry that I didn't catch this earlier.)